src/clojure/nrepl/bencode.clj

```diff
@@ -323,6 +323,16 @@
       (or (nil? thing) (coll? thing) (.isArray (class thing))) :list
       :else (type thing))))
 
+(defn write-bencode*
+  "Similar to `write-bencode`, except it will only write if the whole `thing`
+   is writable. In practice, it avoids sending partial messages, which can
```

Comment:
We should expand on the definition of "writable" here - e.g. with some examples.